### PR TITLE
Ensure that source files contain only valid UTF8 encodings

### PR DESCRIPTION
**Pull request by anson1014**
## Description

Modern software (including text editors, static analysis software, and web-based code review interfaces) often requires source code files to be interpretable via a consistent character encoding, with UTF-8 or ASCII (a strict subset of UTF-8) as the default. Several of the MariaDB source files contain bytes that are not valid in either the UTF-8 or ASCII encodings, but instead represent strings encoded in the ISO-8859-1/Latin-1 or ISO-8859-2/Latin-2 encodings.

These inconsistent encodings may prevent software from correctly presenting or processing such files. Converting all source files to valid UTF8 characters will ensure correct handling.

In summary, the changes in this PR can be categorized into 2 things:

- Comments written in Czech were replaced with lightly-corrected translations from Google Translate.
- Comments describing the proper handling of special characters were changed so that the comments are now purely UTF8. 

## How can this PR be tested?

1. Changes to non-ASCII-encoded bytes in comments have no effect on compiled code.
2. No non-ASCII characters appeared in any other type of code.

## Basing the PR against the correct MariaDB version

* [x] *This is a bug fix and the PR is based against the earliest branch in which the bug can be reproduced*

## Copyright

All new code of the whole pull request, including one or several files that are either new files or modified ones, are contributed under the BSD-new license. I am contributing on behalf of my employer Amazon Web Services, Inc.

### CODE REVIEW

File: strings/ctype-latin1.c
Comment (cvicentiu) [inline strings/ctype-latin1.c:507]: I strongly think that the readability of this comment is greatly affected by not using the proper latin1 character.

File: strings/ctype-latin1.c
Comment (cvicentiu) [inline strings/ctype-latin1.c:517]: Same here. It's sadly much harder to read and I don't think I would easily be able to grok the amended version, had I not had the deleted version for comparison. Same comment for the rest of the file.

File: mysys/my_win_popen.cc
Comment (cvicentiu) [inline mysys/my_win_popen.cc:95]: Ok, this was a typo.

File: storage/connect/domdoc.cpp
Comment (cvicentiu) [inline storage/connect/domdoc.cpp:645]: Ok.

File: storage/connect/frcas.h
Comment (cvicentiu) [inline storage/connect/frcas.h:4]: What if we change these characters to utf-8? Is there a test case that fails? That would make it more consistent with `sql/share/errmsg-utf8.txt`

Comment (cvicentiu) [reply storage/connect/frcas.h:4]: Same comment for all storage/connect files.

File: strings/ctype-czech.c
Comment (cvicentiu) [inline strings/ctype-czech.c:142]: Ok

File: strings/ctype-czech.c
Comment (cvicentiu) [inline strings/ctype-czech.c:130]: Ok

File: strings/ctype-czech.c
Comment (cvicentiu) [inline strings/ctype-czech.c:115]: Ok

File: strings/ctype-czech.c
Comment (cvicentiu) [inline strings/ctype-czech.c:45]: This comment refers to what the file implements. Please don't change it.

File: strings/ctype-czech.c
Comment (cvicentiu) [inline strings/ctype-czech.c:26]: All the "special" characters that are latin-1 encoded are actually in comments. Let's run iconv on this file and convert it to UTF-8. There should be no functional changes.

File: strings/ctype-latin1.c
Comment (cvicentiu) [reply strings/ctype-latin1.c:507]: To get the code consistent, let's format the files with UTF-8 character set as all the "special characters are in comments". In this case it is more important that the reader understand the meaning, rather than using the "latin-1" encoded gliph. We can use the UTF-8 corresponding byte sequence.

File: strings/ctype-czech.c
Comment (cvicentiu) [inline strings/ctype-czech.c:330]: Ok

**Review by cvicentiu — CHANGES_REQUESTED** [review]
Hi @anson1014 !

Thank you for this pull request. I believe we can reach a middle-ground where we encode everything UTF-8 and avoid the ugliness of these escape characters. Please test it with Connect storage engine suite, to make sure we don't introduce any failures there.

File: strings/ctype-czech.c
Comment (dlenski) [reply strings/ctype-czech.c:45]: The existing comment was rather ambiguous: what exactly does "contains" means?

> ```
> This file also contains the ISO-Latin-2 definitions of
> ```

The new version of the comment is much clearer and less ambiguous.

File: strings/ctype-latin1.c
Comment (dlenski) [reply strings/ctype-latin1.c:507]: @cvicentiu, agreed on this one. The easy readability of the comments is important, since they're discussing the properties of certain characters.

@anson1014 and I had discussed the possibility of converting these comments to UTF-8 instead, using `iconv`, but weren't sure which version would be preferred upstream.

File: storage/connect/frcas.h
Comment (dlenski) [reply storage/connect/frcas.h:4]: Without knowing the environment in which these strings are supposed to be _used_, it's very hard to say whether this would work.

Presumably, these strings were encoded in `iso-8859-1` because they were intended to be used as log/console messages in the `fr_FR.iso-8859-1` locale, **not** in the `fr_FR.utf-8` locale. But there's no clear documentation on this point that I can find. :disappointed:

File: strings/ctype-latin1.c
Comment (anson1014) [reply strings/ctype-latin1.c:507]: These characters have been changed to UTF-8

File: strings/ctype-czech.c
Comment (anson1014) [reply strings/ctype-czech.c:26]: The comments in this file have been changed to UTF-8

**Comment by anson1014** [comment]
Changes regarding the encoding of the files under `storage/connect` have been split into a separate [PR](https://github.com/MariaDB/server/pull/2224).

**Comment by ottok** [comment]
@cvicentiu Can you please re-review? New version that addressed all of your feedback has been here for two weeks now.

File: strings/ctype-czech.c
Comment (LinuxJedi) [inline strings/ctype-czech.c:46]: Another really minor thing, please align this with the rest of the comment. I think tabs were used for the rest of this comment block and spaces here.

File: strings/ctype-czech.c
Comment (LinuxJedi) [inline strings/ctype-czech.c:45]: Really minor thing, this line is too long

File: strings/ctype-czech.c
Comment (LinuxJedi) [inline strings/ctype-czech.c:330]: Tabs with this block too please, consistent with the rest of the comment blocks.

**Review by LinuxJedi — CHANGES_REQUESTED** [review]
Pretty much there with this, some nice cleanups. Just some minor changes needed.

**Review by dlenski — APPROVED** [review]

**Review by LinuxJedi — APPROVED** [review]
Great work, many thanks!

**Comment by dlenski** [comment]
I believe the changes [requested](https://github.com/MariaDB/server/pull/2188#pullrequestreview-1042942681) by @cvicentiu last month have all been resolved; _this_ PR now modifies _only_ comments.

(The changes to string literal representation of non-ASCII characters have now been split out into https://github.com/MariaDB/server/pull/2224.)

Could you please review again, @cvicentiu?

**Comment by LinuxJedi** [comment]
@anson1014 can I please request that you click through the CLA system above and tick that you want to contribute via the BSD license (if that is the license you wish to contribute under)? It helps us out a lot.

After that I can merge this. I have spoken to @cvicentiu and it does not require a re-review from him.

Many thanks!

**Comment by ottok** [comment]
@LinuxJedi The CLA system does not allow to choose BSD, therefore we simply have this in all commits:

> All new code of the whole pull request, including one or several files
> that are either new files or modified ones, are contributed under the
> BSD-new license. I am contributing on behalf of my employer
> Amazon Web Services, Inc.

The CLA system on the first page asks to sign the CLA and there is no BSD option, thus **it can't be used by contributors who want to contribute with BSD**.
image

**Comment by LinuxJedi** [comment]
> The CLA system on the first page asks to sign the CLA and there is no BSD option, thus **it can't be used by contributors who want to contribute with BSD**.

@ottok I apologise, I didn't realise we have UX issues with the CLA until after I said that. When you sign in you are then giving the option of using BSD instead. But I agree this is not the most user friendly way of doing it. I'm going to discuss internally to see if we can improve on this. In the mean time, yep, this is fine.

**Pull request merged**
